user-interface/0002-WCON-connect_vega_wallet.md

```diff
@@ -0,0 +1,79 @@
+  - if user rejects connection: **must** see a message saying that the request to connect was denied  (<a name="0002-WCON-015" href="#0002-WCON-015">0002-WCON-015</a>)
+  
+  - if the dapp is unable to connect for technical reason (e.g. CORS): **must** see an explanation of the error, and a method of fixing the issue  (<a name="0002-WCON-016" href="#0002-WCON-016">0002-WCON-016</a>)
+  
+
+- ~~Browser wallet~~ `not available yet`
+  
+- Fairground hosted wallet
+  - **must** only be be shown this option if the dapp is connected to fairground (<a name="0002-WCON-039" href="#0002-WCON-039">0002-WCON-039</a>)
+  - **must** input a wallet name (<a name="0002-WCON-017" href="#0002-WCON-017">0002-WCON-017</a>)
+  - **must** input a password (<a name="0002-WCON-018" href="#0002-WCON-018">0002-WCON-018</a>)
+  - if success: **must** see that the wallet is connected and details of connected key (<a name="0002-WCON-019" href="#0002-WCON-019">0002-WCON-019</a>)
+  - if failure: **must** see reason for failure (<a name="0002-WCON-020" href="#0002-WCON-020">0002-WCON-020</a>)
+  - *note: the fairground hosted wallet is configured to automatically approve connections from dapps so there is no need for key selection.*
+  
+- **must** have the option to select a different method / wallet type if I change my mind (<a name="0002-WCON-021" href="#0002-WCON-021">0002-WCON-021</a>)
+
+... so I can use the interface to read data about my key/party or request my wallet to broadcast transactions to a Vega network.
+
+## Disconnect wallet
+
+When wishing to disconnect my wallet, I...
+
+- **must** see an option to disconnect wallet (<a name="0002-WCON-022" href="#0002-WCON-022">0002-WCON-022</a>)
+- **must** see confirmation that wallet has been disconnected (<a name="0002-WCON-023" href="#0002-WCON-023">0002-WCON-023</a>)
+- **should** see prompt to connect a wallet, after disconnect (<a name="0002-WCON-024" href="#0002-WCON-024">0002-WCON-024</a>)
+
+... so that I can protect my wallet from malicious use or select a different wallet to connect to
+
+
+## Select and switch keys
+
+when looking to do something with a specific key (or set of keys) from my wallet, I...
+
+- **must** see what key is currently selected (if any) (<a name="0002-WCON-025" href="#0002-WCON-025">0002-WCON-025</a>)
+- **must** see a list of keys that are approved from the connected wallet (<a name="0002-WCON-026" href="#0002-WCON-026">0002-WCON-026</a>)
+
+- for each key:
+  - **must** see the first and last 6 digits of the [public key](7001-DATA-data_display.md#public-keys). (<a name="0002-WCON-027" href="#0002-WCON-027">0002-WCON-027</a>)
+  - **should** be able to see the whole public key (<a name="0002-WCON-028" href="#0002-WCON-028">0002-WCON-028</a>)
+  - **must** be able to copy to clipboard the whole public key (<a name="0002-WCON-029" href="#0002-WCON-029">0002-WCON-029</a>)
+  - **must** see the key name/alias (meta data) (<a name="0002-WCON-030" href="#0002-WCON-030">0002-WCON-030</a>)
```

Comment:
Think this must has to go.

Reply:
Because Alias/name is not shared via the currently implemented API?
An alternative would be to have the API updated so that the Alias is shared (other metadata is not).
Or we could comment this out in the assumption that this is coming later.

I think key alias's are really useful for users to be able to keep track of what they have used different keys for (e.g. Diff strategies or isolated margin), It would be a shame for user not to be able to use this between wallet an Dapp.
 
@barnabee do you have a view?

Reply:
See slack for details but it was agreed that name will be shared see https://github.com/vegaprotocol/vegawallet/issues/592 so going to keep this one here